File: scripts/algorithms/C/Coordinate With Maximum Network Quality/Coordinate With Maximum Network Quality.py
class Solution:
    def bestCoordinate(self, towers: List[List[int]], radius: int) -> List[int]:
        return max(
            (
                (sum(qi // (1 + dist) for xi, yi, qi in towers if (dist := sqrt((xi - x) ** 2 + (yi - y) ** 2)) <= radius),
                 [x, y]) for x in range(51) for y in range(51)
            ),
            key=lambda x: (x[0], -x[1][0], -x[1][1])
        )[1]

File: scripts/algorithms/C/Coordinate With Maximum Network Quality/Coordinate With Maximum Network Quality.java
// Runtime: 16 ms (Top 84.85%) | Memory: 41.90 MB (Top 33.33%)

class Solution {
    public int[] bestCoordinate(int[][] towers, int radius) {
        int minX = 51, maxX = 0, minY = 51, maxY = 0, max = 0;
        int[] res = new int[2];
        for(int[] t : towers) {
            minX = Math.min(minX, t[0]);
            maxX = Math.max(maxX, t[0]);
            minY = Math.min(minY, t[1]);
            maxY = Math.max(maxY, t[1]);
        }
        for(int i = minX; i <= maxX; i++) {
            for(int j = minY; j <= maxY; j++) {
                int sum = 0;
                for(int[] t : towers) {
                    int d = (t[0] - i) *(t[0] - i) + (t[1] - j) *(t[1] - j);
                    if(d <= radius * radius) {
                        sum += t[2] /(1+ Math.sqrt(d));    
                    }
                }
                if(sum > max) {
                    max = sum;
                    res = new int[]{i,j};
                }
            }
        }
        return res;
    }
}
